src/commons/asScrollableContent.tsx

```diff
@@ -0,0 +1,82 @@
+import _ from 'lodash';
+import React, {Component} from 'react';
+import {ScrollView, ScrollViewProps} from 'react-native';
+//@ts-ignore
+import hoistStatics from 'hoist-non-react-statics';
+
+interface ScrollableContentProps {
+  scrollableContentProps?: ScrollViewProps;
+  maximumContentSize?: number;
+}
+
+interface ScrollableContentState {
+  maximumContentSize?: number;
+  contentSize?: number;
+  scrollEnabled: boolean;
+}
+
+type PropTypes = ScrollableContentProps;
+
+export default function asScrollableContent(WrappedComponent: React.ComponentType<any>) {
+  class ScrollableContent extends Component<PropTypes, ScrollableContentState> {
+    static displayName: string | undefined;
+
+    constructor(props: PropTypes) {
+      super(props);
+      this.state = {
+        maximumContentSize: props.maximumContentSize,
+        contentSize: undefined,
+        scrollEnabled: false
+      };
+    }
+
+    static isScrollEnabled = (contentSize?: number, maximumContentSize?: number) => {
+      return !_.isUndefined(contentSize) && !_.isUndefined(maximumContentSize) && contentSize > maximumContentSize;
+    };
+
+    static getDerivedStateFromProps(nextProps: any, prevState: any) {
```

Comment:
If you're writing a new component and already has to use `getDerivedStateFromProps` it should already raise a flag. Do we really want this component has to be uncontrolled?